src/components/Home.tsx

```diff
@@ -0,0 +1,163 @@
+import { useQuery } from "@tanstack/react-query";
+import { RankDetails } from "../models/state";
+import { fetchTopManga, fetchTopManhwa, fetchTopManhua } from "../services/api";
+
+function Home() {
+  return (
+    <>
+      <h2 className="text-white font-semibold text-xl">Top Manga</h2>
+      <TopManga />
+      <h2 className="text-white font-semibold text-xl">Top Manhwa</h2>
+      <TopManhwa />
+      <h2 className="text-white font-semibold text-xl">Top Manhua</h2>
+      <TopManhua />
+    </>
+  );
+}
+
+export default Home;
+
+function TopManga() {
+  const { data: topMangaResults, error } = useQuery({
+    queryKey: [`fetchTopManga`],
+    queryFn: () => fetchTopManga(),
+  });
+
+  if (error) {
+    console.error("An unexpected error occurred: " + error);
+  }
+
+  if (!topMangaResults) {
+    return (
+      <div className="top-manga-container w-full h-3/10 text-center">
+        <p className="p-2 font-sans text-white">There is no manga to display</p>
+      </div>
+    );
+  }
+
+  // Filter the results to ensure that there will always be a cover image and narrow it down to the first 10 comics.
+  const topTenManga: RankDetails[] = topMangaResults
+    .filter((comic) => comic.md_covers?.[0]?.b2key)
+    .slice(0, 10);
+
+  return (
+    <div className="top-manga-container w-full h-3/10 carousel carousel-center space-x-4 bg-neutral rounded-box bg-transparent overflow-y-hidden flex align-center py-4 mb-4">
+      {topTenManga.map((mangaRec) => (
+        <div
+          key={mangaRec.slug}
+          className="carousel-item w-2/6 h-full p-0 m-0 flex flex-col justify-center relative indicator first:pl-0 hover:cursor-pointer md:w-1/10 lg:w-1/10"
+        >
+          <span className="indicator-item badge badge-secondary px-2.5 py-3.5 border-none bg-purple-1000 text-white font-semibold font-sans">
+            {topMangaResults.indexOf(mangaRec) + 1}
```

Comment:
So do `{topMangaResults[index + 1]`  instead.

Reply:
Will this not then return a comic object? The only thing I'm doing here is using the index and adding 1 to it (since it starts at 0) to indicate the rank.

Reply:
I see in that case you can do `{index + 1}`

Reply:
Thank you, as I have commented this section of code out for future use, I will keep this in mind.